modules/nginxvts/nginxvts_test.go

```diff
@@ -0,0 +1,261 @@
+			config: Config{
+				HTTP: web.HTTP{
+					Client: web.Client{
+						TLSConfig: tlscfg.TLSConfig{TLSCA: "testdata/tls"},
+					},
+				}},
+		},
+	}
+
+	for name, test := range tests {
+		t.Run(name, func(t *testing.T) {
+			es := New()
+			es.Config = test.config
+
+			if test.wantFail {
+				assert.False(t, es.Init())
+			} else {
+				assert.True(t, es.Init())
+				assert.Equal(t, test.wantNumOfCharts, len(*es.Charts()))
+			}
+		})
+	}
+}
+
+func TestNginxVTS_Check(t *testing.T) {
+	tests := map[string]struct {
+		prepare  func(*testing.T) (vts *NginxVTS, cleanup func())
+		wantFail bool
+	}{
+		"valid data":         {prepare: prepareNginxVTSValidData},
+		"invalid data":       {prepare: prepareNginxVTSInvalidData, wantFail: true},
+		"404":                {prepare: prepareNginxVTS404, wantFail: true},
+		"connection refused": {prepare: prepareNginxVTSConnectionRefused, wantFail: true},
+	}
+
+	for name, test := range tests {
+		t.Run(name, func(t *testing.T) {
+			vts, cleanup := test.prepare(t)
+			defer cleanup()
+
+			if test.wantFail {
+				assert.False(t, vts.Check())
+			} else {
+				assert.True(t, vts.Check())
+			}
+		})
+	}
+}
+
+func TestNginxVTS_Charts(t *testing.T) {
+	assert.Nil(t, New().Charts())
+}
+
+func TestNginxVTS_Cleanup(t *testing.T) {
+	assert.NotPanics(t, New().Cleanup)
+}
+
+func TestNginxVTS_Collect(t *testing.T) {
+	tests := map[string]struct {
+		// prepare       func() *NginxVTS
+		prepare       func(t *testing.T) (vts *NginxVTS, cleanup func())
+		wantCollected map[string]int64
+		checkCharts   bool
+	}{
+		"right metrics": {
+			prepare: prepareNginxVTSValidData,
+			wantCollected: map[string]int64{
+				// Nginx running time
+				"loadmsec": 1606489796895,
+				"nowmsec":  1606490116734,
+				// Nginx connections
+				"connections_active":   2,
+				"connections_reading":  0,
+				"connections_writing":  1,
+				"connections_waiting":  1,
+				"connections_accepted": 12,
+				"connections_handled":  12,
+				"connections_total":    17,
+				// Nginx shared memory
+				"sharedzones_maxsize":  1048575,
+				"sharedzones_usedsize": 45799,
+				"sharedzones_usednode": 13,
+				// Nginx traffic
+				"total_requestcounter": 2,
+				"total_inbytes":        156,
+				"total_outbytes":       692,
+				// Nginx response code
+				"total_responses_1xx": 1,
+				"total_responses_2xx": 2,
+				"total_responses_3xx": 3,
+				"total_responses_4xx": 4,
+				"total_responses_5xx": 5,
+				// Nginx cache
+				"total_cache_miss":        2,
+				"total_cache_bypass":      4,
+				"total_cache_expired":     6,
+				"total_cache_stale":       8,
+				"total_cache_updating":    10,
+				"total_cache_revalidated": 12,
+				"total_cache_hit":         14,
+				"total_cache_scarce":      16,
+			},
+			checkCharts: true,
+		},
+	}
+
+	for name, test := range tests {
+		t.Run(name, func(t *testing.T) {
+			vts, cleanup := test.prepare(t)
+			defer cleanup()
+
+			var collected map[string]int64
+			collected = vts.Collect()
```

Comment:
```suggestion
			collected := vts.Collect()
```

see reviewdog warnings